internal/dataplane/parser/translators/ingress_atc.go

```diff
@@ -0,0 +1,243 @@
+package translators
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/kong/go-kong/kong"
+	netv1 "k8s.io/api/networking/v1"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/annotations"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/kongstate"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/parser/atc"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/util"
+)
+
+var (
+	headerAnnotationRegexPrefix = "~*"
+	// TODO: list all available matchers instead.
+	validMethods = regexp.MustCompile(`\A[A-Z]+$`)
+
+	// hostnames are complicated. shamelessly cribbed from https://stackoverflow.com/a/18494710
+	// TODO if the Kong core adds support for wildcard SNI route match criteria, this should change.
```

Comment:
Ditto

Reply:
I think we should put this to GH issue when kong support this.